internal/tofu/node_variable_reference.go

```diff
@@ -146,6 +146,7 @@ func (n *nodeVariableReferenceInstance) ModulePath() addrs.Module {
 func (n *nodeVariableReferenceInstance) Execute(ctx EvalContext, op walkOperation) tfdiags.Diagnostics {
 	log.Printf("[TRACE] nodeVariableReferenceInstance: evaluating %s", n.Addr)
 	diags := evalVariableValidations(n.Addr, n.Config, n.Expr, ctx)
+	diags = diags.Append(evalVariableDeprecation(n.Addr, n.Config, n.Expr, ctx, false)) // TODO provide this bool from cli arg
```

Comment:
I believe it should work for `tofu validate` as well. Could you please confirm with a manual or auto test?

Reply:
Yes, it works well.
![Screenshot 2025-03-07 at 11 23 57](https://github.com/user-attachments/assets/d20ec915-eab2-4ef6-99dd-9a3dd8eb960b)

Reply:
Thanks for checking it!